src/D2L.Security.OAuth2/Validation/AccessTokens/AccessTokenValidator.cs

```diff
@@ -61,9 +61,10 @@ string token
 
 			string keyId = unvalidatedToken.Header[ "kid" ].ToString();
 
-			D2LSecurityToken signingKey = await m_publicKeyProvider
+			using D2LSecurityToken signingKey = ( await m_publicKeyProvider
 				.GetByIdAsync( keyId )
-				.ConfigureAwait( false );
+				.ConfigureAwait( false )
+			).Ref();

```

Comment:
Need to `Ref()` and dispose here now so the cache disposing doesn't fully dispose it during usage.